Intermittent test failure: TestInternalEventStreamDeliveryAtHeadWithSourceAddress

See https://github.com/kaleido-io/paladin/actions/runs/10873302269/job/30169263447?pr=154

<details>
<summary>View log snippet</summary>

```
--- FAIL: TestInternalEventStreamDeliveryAtHeadWithSourceAddress (0.04s)
    event_streams_test.go:186: 
        	Error Trace:	/home/runner/work/paladin/paladin/core/go/pkg/blockindexer/event_streams_test.go:186
        	Error:      	Not equal: 
        	            	expected: map[string]interface {}{"intParam1":"1000001", "strParam2":"event_b_in_block_1"}
        	            	actual  : map[string]interface {}{"structParam1":map[string]interface {}{"intArrayField":[]interface {}{"1000", "2000", "3000", "4000", "5000"}, "strField":"event_c_in_block_0"}}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,12 @@
        	            	-(map[string]interface {}) (len=2) {
        	            	- (string) (len=9) "intParam1": (string) (len=7) "1000001",
        	            	- (string) (len=9) "strParam2": (string) (len=18) "event_b_in_block_1"
        	            	+(map[string]interface {}) (len=1) {
        	            	+ (string) (len=12) "structParam1": (map[string]interface {}) (len=2) {
        	            	+  (string) (len=13) "intArrayField": ([]interface {}) (len=5) {
        	            	+   (string) (len=4) "1000",
        	            	+   (string) (len=4) "2000",
        	            	+   (string) (len=4) "3000",
        	            	+   (string) (len=4) "4000",
        	            	+   (string) (len=4) "5000"
        	            	+  },
        	            	+  (string) (len=8) "strField": (string) (len=18) "event_c_in_block_0"
        	            	+ }
        	            	 }
        	Test:       	TestInternalEventStreamDeliveryAtHeadWithSourceAddress
    event_streams_test.go:186: 
        	Error Trace:	/home/runner/work/paladin/paladin/core/go/pkg/blockindexer/event_streams_test.go:186
        	Error:      	Not equal: 
        	            	expected: map[string]interface {}{"intParam1":"1000002", "strParam2":"event_b_in_block_2"}
        	            	actual  : map[string]interface {}{"intParam1":"1000001", "strParam2":"event_b_in_block_1"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,4 @@
        	            	 (map[string]interface {}) (len=2) {
        	            	- (string) (len=9) "intParam1": (string) (len=7) "1000002",
        	            	- (string) (len=9) "strParam2": (string) (len=18) "event_b_in_block_2"
        	            	+ (string) (len=9) "intParam1": (string) (len=7) "1000001",
        	            	+ (string) (len=9) "strParam2": (string) (len=18) "event_b_in_block_1"
        	            	 }
        	Test:       	TestInternalEventStreamDeliveryAtHeadWithSourceAddress
    event_streams_test.go:186: 
        	Error Trace:	/home/runner/work/paladin/paladin/core/go/pkg/blockindexer/event_streams_test.go:186
        	Error:      	Not equal: 
        	            	expected: map[string]interface {}{"intParam1":"1000003", "strParam2":"event_b_in_block_3"}
        	            	actual  : map[string]interface {}{"structParam1":map[string]interface {}{"intArrayField":[]interface {}{"1001", "2001", "3001", "4001", "5001"}, "strField":"event_c_in_block_1"}}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,12 @@
        	            	-(map[string]interface {}) (len=2) {
        	            	- (string) (len=9) "intParam1": (string) (len=7) "1000003",
        	            	- (string) (len=9) "strParam2": (string) (len=18) "event_b_in_block_3"
        	            	+(map[string]interface {}) (len=1) {
        	            	+ (string) (len=12) "structParam1": (map[string]interface {}) (len=2) {
        	            	+  (string) (len=13) "intArrayField": ([]interface {}) (len=5) {
        	            	+   (string) (len=4) "1001",
        	            	+   (string) (len=4) "2001",
        	            	+   (string) (len=4) "3001",
        	            	+   (string) (len=4) "4001",
        	            	+   (string) (len=4) "5001"
        	            	+  },
        	            	+  (string) (len=8) "strField": (string) (len=18) "event_c_in_block_1"
        	            	+ }
        	            	 }
        	Test:       	TestInternalEventStreamDeliveryAtHeadWithSourceAddress
    event_streams_test.go:186: 
        	Error Trace:	/home/runner/work/paladin/paladin/core/go/pkg/blockindexer/event_streams_test.go:186
        	Error:      	Not equal: 
        	            	expected: map[string]interface {}{"intParam1":"1000004", "strParam2":"event_b_in_block_4"}
        	            	actual  : map[string]interface {}{"intParam1":"1000002", "strParam2":"event_b_in_block_2"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,4 @@
        	            	 (map[string]interface {}) (len=2) {
        	            	- (string) (len=9) "intParam1": (string) (len=7) "1000004",
        	            	- (string) (len=9) "strParam2": (string) (len=18) "event_b_in_block_4"
        	            	+ (string) (len=9) "intParam1": (string) (len=7) "1000002",
        	            	+ (string) (len=9) "strParam2": (string) (len=18) "event_b_in_block_2"
        	            	 }
        	Test:       	TestInternalEventStreamDeliveryAtHeadWithSourceAddress
    event_streams_test.go:186: 
        	Error Trace:	/home/runner/work/paladin/paladin/core/go/pkg/blockindexer/event_streams_test.go:186
        	Error:      	Not equal: 
        	            	expected: map[string]interface {}{"intParam1":"1000005", "strParam2":"event_b_in_block_5"}
        	            	actual  : map[string]interface {}{"intParam1":"1000003", "strParam2":"event_b_in_block_3"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,4 @@
        	            	 (map[string]interface {}) (len=2) {
        	            	- (string) (len=9) "intParam1": (string) (len=7) "1000005",
        	            	- (string) (len=9) "strParam2": (string) (len=18) "event_b_in_block_5"
        	            	+ (string) (len=9) "intParam1": (string) (len=7) "1000003",
        	            	+ (string) (len=9) "strParam2": (string) (len=18) "event_b_in_block_3"
        	            	 }
        	Test:       	TestInternalEventStreamDeliveryAtHeadWithSourceAddress
    event_streams_test.go:186: 
        	Error Trace:	/home/runner/work/paladin/paladin/core/go/pkg/blockindexer/event_streams_test.go:186
        	Error:      	Not equal: 
        	            	expected: map[string]interface {}{"intParam1":"1000006", "strParam2":"event_b_in_block_6"}
        	            	actual  : map[string]interface {}{"intParam1":"1000004", "strParam2":"event_b_in_block_4"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,4 @@
        	            	 (map[string]interface {}) (len=2) {
        	            	- (string) (len=9) "intParam1": (string) (len=7) "1000006",
        	            	- (string) (len=9) "strParam2": (string) (len=18) "event_b_in_block_6"
        	            	+ (string) (len=9) "intParam1": (string) (len=7) "1000004",
        	            	+ (string) (len=9) "strParam2": (string) (len=18) "event_b_in_block_4"
        	            	 }
        	Test:       	TestInternalEventStreamDeliveryAtHeadWithSourceAddress
    event_streams_test.go:186: 
        	Error Trace:	/home/runner/work/paladin/paladin/core/go/pkg/blockindexer/event_streams_test.go:186
        	Error:      	Not equal: 
        	            	expected: map[string]interface {}{"intParam1":"1000007", "strParam2":"event_b_in_block_7"}
        	            	actual  : map[string]interface {}{"intParam1":"1000005", "strParam2":"event_b_in_block_5"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,4 @@
        	            	 (map[string]interface {}) (len=2) {
        	            	- (string) (len=9) "intParam1": (string) (len=7) "1000007",
        	            	- (string) (len=9) "strParam2": (string) (len=18) "event_b_in_block_7"
        	            	+ (string) (len=9) "intParam1": (string) (len=7) "1000005",
        	            	+ (string) (len=9) "strParam2": (string) (len=18) "event_b_in_block_5"
        	            	 }
        	Test:       	TestInternalEventStreamDeliveryAtHeadWithSourceAddress
    event_streams_test.go:186: 
        	Error Trace:	/home/runner/work/paladin/paladin/core/go/pkg/blockindexer/event_streams_test.go:186
        	Error:      	Not equal: 
        	            	expected: map[string]interface {}{"intParam1":"1000008", "strParam2":"event_b_in_block_8"}
        	            	actual  : map[string]interface {}{"intParam1":"1000006", "strParam2":"event_b_in_block_6"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,4 @@
        	            	 (map[string]interface {}) (len=2) {
        	            	- (string) (len=9) "intParam1": (string) (len=7) "1000008",
        	            	- (string) (len=9) "strParam2": (string) (len=18) "event_b_in_block_8"
        	            	+ (string) (len=9) "intParam1": (string) (len=7) "1000006",
        	            	+ (string) (len=9) "strParam2": (string) (len=18) "event_b_in_block_6"
        	            	 }
        	Test:       	TestInternalEventStreamDeliveryAtHeadWithSourceAddress
    event_streams_test.go:186: 
        	Error Trace:	/home/runner/work/paladin/paladin/core/go/pkg/blockindexer/event_streams_test.go:186
        	Error:      	Not equal: 
        	            	expected: map[string]interface {}{"intParam1":"1000009", "strParam2":"event_b_in_block_9"}
        	            	actual  : map[string]interface {}{"intParam1":"1000007", "strParam2":"event_b_in_block_7"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,4 @@
        	            	 (map[string]interface {}) (len=2) {
        	            	- (string) (len=9) "intParam1": (string) (len=7) "1000009",
        	            	- (string) (len=9) "strParam2": (string) (len=18) "event_b_in_block_9"
        	            	+ (string) (len=9) "intParam1": (string) (len=7) "1000007",
        	            	+ (string) (len=9) "strParam2": (string) (len=18) "event_b_in_block_7"
        	            	 }
        	Test:       	TestInternalEventStreamDeliveryAtHeadWithSourceAddress
    event_streams_test.go:186: 
        	Error Trace:	/home/runner/work/paladin/paladin/core/go/pkg/blockindexer/event_streams_test.go:186
        	Error:      	Not equal: 
        	            	expected: map[string]interface {}{"intParam1":"1000010", "strParam2":"event_b_in_block_10"}
        	            	actual  : map[string]interface {}{"intParam1":"1000008", "strParam2":"event_b_in_block_8"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,4 @@
        	            	 (map[string]interface {}) (len=2) {
        	            	- (string) (len=9) "intParam1": (string) (len=7) "1000010",
        	            	- (string) (len=9) "strParam2": (string) (len=19) "event_b_in_block_10"
        	            	+ (string) (len=9) "intParam1": (string) (len=7) "1000008",
        	            	+ (string) (len=9) "strParam2": (string) (len=18) "event_b_in_block_8"
        	            	 }
        	Test:       	TestInternalEventStreamDeliveryAtHeadWithSourceAddress
    event_streams_test.go:186: 
        	Error Trace:	/home/runner/work/paladin/paladin/core/go/pkg/blockindexer/event_streams_test.go:186
        	Error:      	Not equal: 
        	            	expected: map[string]interface {}{"intParam1":"1000011", "strParam2":"event_b_in_block_11"}
        	            	actual  : map[string]interface {}{"intParam1":"1000009", "strParam2":"event_b_in_block_9"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,4 @@
        	            	 (map[string]interface {}) (len=2) {
        	            	- (string) (len=9) "intParam1": (string) (len=7) "1000011",
        	            	- (string) (len=9) "strParam2": (string) (len=19) "event_b_in_block_11"
        	            	+ (string) (len=9) "intParam1": (string) (len=7) "1000009",
        	            	+ (string) (len=9) "strParam2": (string) (len=18) "event_b_in_block_9"
        	            	 }
        	Test:       	TestInternalEventStreamDeliveryAtHeadWithSourceAddress
    event_streams_test.go:186: 
        	Error Trace:	/home/runner/work/paladin/paladin/core/go/pkg/blockindexer/event_streams_test.go:186
        	Error:      	Not equal: 
        	            	expected: map[string]interface {}{"intParam1":"1000012", "strParam2":"event_b_in_block_12"}
        	            	actual  : map[string]interface {}{"intParam1":"1000010", "strParam2":"event_b_in_block_10"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,4 @@
        	            	 (map[string]interface {}) (len=2) {
        	            	- (string) (len=9) "intParam1": (string) (len=7) "1000012",
        	            	- (string) (len=9) "strParam2": (string) (len=19) "event_b_in_block_12"
        	            	+ (string) (len=9) "intParam1": (string) (len=7) "1000010",
        	            	+ (string) (len=9) "strParam2": (string) (len=19) "event_b_in_block_10"
        	            	 }
        	Test:       	TestInternalEventStreamDeliveryAtHeadWithSourceAddress
    event_streams_test.go:186: 
        	Error Trace:	/home/runner/work/paladin/paladin/core/go/pkg/blockindexer/event_streams_test.go:186
        	Error:      	Not equal: 
        	            	expected: map[string]interface {}{"intParam1":"1000013", "strParam2":"event_b_in_block_13"}
        	            	actual  : map[string]interface {}{"intParam1":"1000011", "strParam2":"event_b_in_block_11"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,4 @@
        	            	 (map[string]interface {}) (len=2) {
        	            	- (string) (len=9) "intParam1": (string) (len=7) "1000013",
        	            	- (string) (len=9) "strParam2": (string) (len=19) "event_b_in_block_13"
        	            	+ (string) (len=9) "intParam1": (string) (len=7) "1000011",
        	            	+ (string) (len=9) "strParam2": (string) (len=19) "event_b_in_block_11"
        	            	 }
        	Test:       	TestInternalEventStreamDeliveryAtHeadWithSourceAddress
    event_streams_test.go:186: 
        	Error Trace:	/home/runner/work/paladin/paladin/core/go/pkg/blockindexer/event_streams_test.go:186
        	Error:      	Not equal: 
        	            	expected: map[string]interface {}{"intParam1":"1000014", "strParam2":"event_b_in_block_14"}
        	            	actual  : map[string]interface {}{"intParam1":"1000012", "strParam2":"event_b_in_block_12"}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,4 +1,4 @@
        	            	 (map[string]interface {}) (len=2) {
        	            	- (string) (len=9) "intParam1": (string) (len=7) "1000014",
        	            	- (string) (len=9) "strParam2": (string) (len=19) "event_b_in_block_14"
        	            	+ (string) (len=9) "intParam1": (string) (len=7) "1000012",
        	            	+ (string) (len=9) "strParam2": (string) (len=19) "event_b_in_block_12"
        	            	 }
        	Test:       	TestInternalEventStreamDeliveryAtHeadWithSourceAddress
```
</details>

--- a/core/go/pkg/blockindexer/block_indexer.go
+++ b/core/go/pkg/blockindexer/block_indexer.go
@@ -825,7 +825,7 @@ func (bi *blockIndexer) DecodeTransactionEvents(ctx context.Context, hash tktype
 	for i, event := range events {
 		decoded[i] = &EventWithData{IndexedEvent: event}
 	}
-	err = bi.enrichTransactionEvents(ctx, abi, hash, decoded, false /* no retry */)
+	err = bi.enrichTransactionEvents(ctx, abi, nil, hash, decoded, false /* no retry */)
 	return decoded, err
 }
 
@@ -841,7 +841,7 @@ func (bi *blockIndexer) getConfirmedTransactionReceipt(ctx context.Context, tx e
 	return receipt, nil
 }
 
-func (bi *blockIndexer) enrichTransactionEvents(ctx context.Context, abi abi.ABI, tx tktypes.Bytes32, events []*EventWithData, indefiniteRetry bool) error {
+func (bi *blockIndexer) enrichTransactionEvents(ctx context.Context, abi abi.ABI, source *tktypes.EthAddress, tx tktypes.Bytes32, events []*EventWithData, indefiniteRetry bool) error {
 	// Get the TX receipt with all the logs
 	var receipt *TXReceiptJSONRPC
 	err := bi.retry.Do(ctx, func(attempt int) (_ bool, err error) {
@@ -857,7 +857,7 @@ func (bi *blockIndexer) enrichTransactionEvents(ctx context.Context, abi abi.ABI
 		for _, e := range events {
 			if ethtypes.HexUint64(e.LogIndex) == l.LogIndex {
 				// This the the log for this event - try and enrich the .Data field
-				_ = bi.matchLog(ctx, abi, l, e, nil)
+				_ = bi.matchLog(ctx, abi, l, e, source)
 				break
 			}
 		}

--- a/core/go/pkg/blockindexer/event_streams.go
+++ b/core/go/pkg/blockindexer/event_streams.go
@@ -583,5 +583,5 @@ func (es *eventStream) processCatchupEventPage(checkpointBlock int64, catchUpToB
 }
 
 func (es *eventStream) queryTransactionEvents(tx tktypes.Bytes32, events []*EventWithData, done chan error) {
-	done <- es.bi.enrichTransactionEvents(es.ctx, es.eventABIs, tx, events, true /* retry indefinitely */)
+	done <- es.bi.enrichTransactionEvents(es.ctx, es.eventABIs, es.definition.Source, tx, events, true /* retry indefinitely */)
 }

